### ISSUE TITLE
lighthousebot not failing PRs when score < minScore

### ISSUE THREAD

**Issue by ankeetmaini** [ebidel/lighthouse-ci#8]
I debugged this by running the frontend server in my local and the reason is when I add lighthousebot as a collaborator, the request isn't accepted by the bot and hence it doesn't have the rights and fails. It can only post a comment.

I even checked github api to add a webhook and a handler in server to auto accept requests but at this time there isn't a way we can automate this process.

**Comment by ebidel** [comment]
Collaborators should be able to post comments on repos. Can you provide any more details on the failures? The workflow doesn't work for webhook, so no need to change any settings there.

**Comment by ankeetmaini** [comment]
No, I created my own `.oauth_token`, it's just that if I don't accept the collaborator request github api doesn't allow to change the status.

See this PR -> https://github.com/ankeetmaini/react-hn/pull/11, here your bot posts a comment but fails to mark the PR as a failure.

The one I tried with my own `.oauth_token` passes because I accepted the collaborator request from my bot account visible in the screenshot `perf-bot` https://github.com/ankeetmaini/react-hn/pull/9
<img width="985" alt="screen shot 2017-09-18 at 10 17 13 pm" src="https://user-images.githubusercontent.com/6652823/30553585-4e5582b2-9cbf-11e7-82eb-6a853f6f7c26.png">

**Comment by ankeetmaini** [comment]
So I'm guessing that the lighthousebot account needs to accept collaborator request from all the repos. Correct me if I am wrong?

**Comment by ebidel** [comment]
I see what you mean! Didn't realize there was an invitation link to collaborate on repos. I accepted your invite.

Hmm, do you know of a way to work around that? Email invites are not going to scale well :(

**Comment by ankeetmaini** [comment]
I checked the GitHub API if we can add a handler in the front end server to
auto accept any invite which we can hook into via git webhooks. But at this
time there doesn't exist an API. I can raise a feature request but not sure
if it'll be taken up!

Or we can ask repository owners to give their .oauth_token and use that? In
a set killer you're doing for the webpage test request id? But it'll grow
with time, might make server slow and we'd need to move this to a redis
client.

**Comment by ebidel** [comment]
An `Authorization` header with the user's oauth token would be great, but that would that work?
 Wouldn't that also post as the user and not the lighthouse bot profile icon?

**Comment by ankeetmaini** [comment]
Yes the moment we depend upon the user's oauth it'll be posted as his
comment. We can add the icon in the comment body? :-D

**Comment by ankeetmaini** [comment]
Also we can save the user's . oauth_token in a secret environment variable
which the lighthouse-ci will be able to access and send to the frontend
server!

**Comment by ebidel** [comment]
Those might be an option. Without a proper Github Integration, I do like that users see the official name and icon:

<img width="852" alt="27059055-70ba6e86-4f89-11e7-8ead-932aab0f2634" src="https://user-images.githubusercontent.com/238208/30558639-029c4eda-9c67-11e7-9f94-8ae72a92796c.png">

Otherwise, they're going to see themselves and their name. Could be confusing. 

Right now the volume of collaborator emails is low. I was just ignoring them :(

**Comment by ankeetmaini** [comment]
Let me know if I can help in resolving this!

**Comment by schoenwaldnils** [comment]
Hi, I'm having the same issue.
https://travis-ci.org/schoenwaldnils/timeline/builds/287505971
https://github.com/schoenwaldnils/timeline/pull/1

The comment is there, but I don't get a failing test.

**Comment by ebidel** [comment]
@schoenwaldnils That setup looks good to me. You've got the bot as a collaborator on the repo and using an API as the Travis env https://github.com/ebidel/lighthouse-ci#get-an-api-key?

**Comment by schoenwaldnils** [comment]
It works now. I might had to wait until @lighthousebot accepts the collaborator-request.

**Comment by riderx** [comment]
3 days and lighthouse still didn't accept :/ how can i change that ?

**Comment by EoinTraynor** [comment]
This is obviously still an ongoing issue. 
What is the expected duration for lighthouse to accept my collaborator-request? It's been 6 days since I have requested it.

**Comment by krzkaczor** [comment]
If anyone still needs it [lighthouse-keeper](https://github.com/codechecks/lighthouse-keeper) supports failing PRs if score is too low, as well as tracking score changes between PRs and many other things. Cheers!
